### PR TITLE
Closes #1471: Add support for text prompts dialogs.

### LINKED ISSUE #1471: Add support for js prompt dialogs which ask  user to enter a message: GeckoEngine
User Story: https://github.com/mozilla-mobile/reference-browser/issues/223

### PR DESCRIPTION
**Pull request by Amejia481**
Added support for [Window.prompt()](https://developer.mozilla.org/en-US/docs/Web/API/Window/prompt#Parameters)

[Demo](https://drive.google.com/open?id=1wgbhk8qzgluGMHVDVP5Cx6-CZUnk_5-u)

### CODE REVIEW

File: docs/changelog.md [hunk 1]
```diff
@@ -47,6 +50,7 @@ permalink: /changelog/
 
 * **feature-prompts**, **browser-engine-gecko***
   * Added support for Authentication dialogs.
+  * Added support for [datetime-local](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/datetime-local) and [time](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/time) pickers.
```
Comment (cadeyrn) [inline docs/changelog.md]: you duplicated this changelog entry. ;-)

Comment (Amejia481) [reply docs/changelog.md]: Oops 🙃  Thanks for noticing it :)

Comment (Amejia481) [reply docs/changelog.md]: Fixed!

File: components/feature/prompts/src/main/java/mozilla/components/feature/prompts/TextPromptDialogFragment.kt [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+            .setTitle(title)
+            .setCancelable(true)
+            .setPositiveButton(android.R.string.ok) { _, _ ->
+                onPositiveClickAction()
+            }
+        return addLayout(builder).create()
+    }
+
+    override fun onCancel(dialog: DialogInterface?) {
+        super.onCancel(dialog)
```
Comment (jonalmeida) [inline components/feature/prompts/src/main/java/mozilla/components/feature/prompts/TextPromptDialogFragment.kt]: Do we need to call super?

**Pull request merged**